flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -342,6 +346,14 @@ private ContainerLaunchContext createTaskExecutorLaunchContext(Resource resource
 		FiniteDuration teRegistrationTimeout = new FiniteDuration(timeout, TimeUnit.SECONDS);
 		final Configuration taskManagerConfig = BootstrapTools.generateTaskManagerConfiguration(
 				flinkConfig, "", 0, 1, teRegistrationTimeout);
+		// Add resource profile of slots to task executor config. 
+		// For yarn, all slots in a task executor have same resource profile 
+		ByteArrayOutputStream output = new ByteArrayOutputStream();
+		ObjectOutputStream rpOutput = new ObjectOutputStream(output);
+		rpOutput.writeObject(getResourceProfile(priority.getPriority()));
+		rpOutput.close();
+		taskManagerConfig.setString(ConfigConstants.TASK_MANAGER_RESOURCE_PROFILE_KEY,
+				new String(Base64.encodeBase64(output.toByteArray())));
```

Comment:
I think we shouldn't base64 encode the resource profile into the `taskManagerConfig`. Instead use `InstantiationUtil.writeObjectToConfig` to write serialized data to the configuration. 

But I'm a little bit torn apart here because so far we used to transfer this kind of information via the environment variables. Maybe @rmetzger can chime in to say what the most idiomatic way to transfer TM data would be.